### PR TITLE
Refactor(eos_cli_config_gen): Wildcard dict to list for ip_extcommunity_lists

### PR DESCRIPTION
**Pull request by KonikaChaurasiya-GSLab**
## Data model key

`ip_extcommunity_lists`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update all `host_vars` under molecule scenario `eos_cli_config_gen_v4.0` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Run molecule `cd ansible_collections/arista/avd/molecule ; make cli-4.0-schema`
  - [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Tony
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

**Comment by ankudinov** [comment]
Please take a look at extended ACL data model and this issue as a possible reference for the data model: https://github.com/aristanetworks/ansible-avd/issues/203

**Comment by ankudinov** [comment]
I suggest at leadt following modifications:
- replace "types" with "entries"
- replace "type" with "action"
- I'd also replace "extcommunities" with something like ext_community_list and make it a list

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md [hunk 1]
```diff
@@ -657,12 +657,14 @@ community_lists:
...
 ```yaml
 ip_extcommunity_lists:
-  < community_list_name_1 >:
-    - type: < permit | deny >
-      extcommunities: "< communities as string >"
-  < community_list_name_2 >:
-    - type: < permit | deny >
-      extcommunities: "< communities as string >"
+  - name: < community_list_name_1 >
+    types:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]: change to `entries`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-extended-community-lists.j2 [hunk 2]
```diff
@@ -6,10 +6,10 @@
 
 | List Name | Type | Extended Communities |
 | --------- | ---- | -------------------- |
-{%     for ip_extcommunity_list in ip_extcommunity_lists | arista.avd.natural_sort %}
-{%         for sequence in ip_extcommunity_lists[ip_extcommunity_list] | arista.avd.default([]) %}
-{%             if sequence.type is arista.avd.defined and sequence.extcommunities is arista.avd.defined %}
-| {{ ip_extcommunity_list }} | {{ sequence.type }} | {{ sequence.extcommunities }} |
+{%     for ip_extcommunity_list in ip_extcommunity_lists | arista.avd.convert_dicts('name') | arista.avd.natural_sort('name') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-extended-community-lists.j2]: add secondary key

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-extended-community-lists.j2 [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {# eos - ip-extended-community-lists #}
-{% for ip_extcommunity_list in ip_extcommunity_lists | arista.avd.natural_sort %}
+{% for ip_extcommunity_list in ip_extcommunity_lists | arista.avd.convert_dicts('name') | arista.avd.natural_sort('name') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-extended-community-lists.j2]: add secondary key

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**